seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/resourcemanager/ResourceRequestHandler.java

```diff
@@ -222,7 +250,53 @@ public Optional<WorkerProfile> preCheckWorkerResource(ResourceProfile r) {
                                                         slot ->
                                                                 slot.getResourceProfile()
                                                                         .enoughThan(r)))
-                        .findAny();
+                        .collect(Collectors.toList());
+
+        Optional<WorkerProfile> workerProfile;
+        switch (allocateStrategy) {
+            case SYSTEM_LOAD:
```

Comment:
each strategy use a class to maintain pick logic.

Reply:
Already modified